File: app/src/main/java/com/karla/learningverbs/viewmodel/VerbDetailViewModel.java
package com.karla.learningverbs.viewmodel;


import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.karla.learningverbs.model.Verb;
import com.karla.learningverbs.repository.userrepository.UserRepository2;
import com.karla.learningverbs.repository.verblist.VerbRepository;
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel;
import com.karla.learningverbs.utils.firebase.CustomEventListener;
import com.google.android.gms.tasks.OnSuccessListener;

public class VerbDetailViewModel extends BaseViewModel {

    private MutableLiveData<Boolean> isVerbFavorite = new MutableLiveData<>();
    private VerbRepository verbDetailRepository;
    private UserRepository2 userRepository2;

    public VerbDetailViewModel() {
        verbDetailRepository = VerbRepository.getInstance();
        userRepository2 = UserRepository2.getInstance();
    }

    public void responseVerbFavoriteUser(String verbId) {
        verbDetailRepository.getUserVerbFavorite(userRepository2.getUserId(), verbId, new CustomEventListener<Verb>(Verb.class) {
            @Override
            public void onSuccess(Verb response) {
                isVerbFavorite.setValue(response != null);
            }

            @Override
            public void onFailed(Throwable throwable) {
                isVerbFavorite.setValue(false);
            }

            @Override
            public void showLoaging() {
                loading.setValue(true);
            }

            @Override
            public void hideLoading() {
                loading.setValue(false);
            }
        });
    }

    public void deleteelement(String verbid) {
        verbDetailRepository.removeUserVerbFavorite(userRepository2.getUserId(), verbid, new OnSuccessListener() {
            @Override
            public void onSuccess(Object o) {
            }
        });
    }

    public void fillDb(Verb verb, String verbId) {
        verbDetailRepository.addVerbFavoriteUser(userRepository2.getUserId(), verbId, verb);
    }

    public LiveData<Boolean> getIsFavoriteVerb() {
        return isVerbFavorite;
    }
}

File: app/src/main/java/com/karla/learningverbs/kotlin/viewmodel/VerbDetailViewModel.kt
package com.karla.learningverbs.kotlin.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.google.android.gms.tasks.OnSuccessListener
import com.karla.learningverbs.kotlin.repository.userrepository.UserRepository
import com.karla.learningverbs.kotlin.repository.verblist.VerbRepository
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel
import com.karla.learningverbs.kotlin.utils.firebase.CustomEventListener
import com.karla.learningverbs.model.Verb

class VerbDetailViewModel: BaseViewModel() {
    private val isVerbFavorite = MutableLiveData<Boolean>()
    private val verbDetailRepository: VerbRepository = VerbRepository()
    private val userRepository: UserRepository = UserRepository()

    fun responseVerbFavoriteUser(verbId: String?) {
        verbDetailRepository.getUserVerbFavorite(
            userRepository.getUserId(),
            verbId, object : CustomEventListener<Verb>(Verb::class.java) {

                override fun onSuccess(response: Verb) {
                    isVerbFavorite.value = response != null
                }

                override fun onFailed(throwable: Throwable?) {
                    isVerbFavorite.value = false
                }

                override fun showLoaging() {
                    loading.value = true
                }

                override fun hideLoading() {
                    loading.value = false
                }
            })
    }

    fun deleteelement(verbid: String?) {
        verbDetailRepository!!.removeUserVerbFavorite(userRepository!!.getUserId(), verbid,
            OnSuccessListener<Void> {

            })
    }

    fun fillDb(verb: Verb?, verbId: String?) {
        verbDetailRepository!!.addVerbFavoriteUser(userRepository!!.getUserId(), verbId, verb)
    }

    fun getIsFavoriteVerb(): LiveData<Boolean> {
        return isVerbFavorite
    }
}